lib/fizzy/execute.hpp

```diff
@@ -100,6 +100,20 @@ execution_result execute(
 // TODO: remove this helper
 execution_result execute(const Module& module, FuncIdx func_idx, std::vector<uint64_t> args);
 
+// Function that should be used by instantiate as imports, identified by module and function name.
+struct ImportedFunction
+{
+    std::string module;
+    std::string name;
+    std::function<execution_result(Instance&, std::vector<uint64_t>, int depth)> function;
```

Comment:
I think `ImportedFunction` should have two more fields:
- `std::vector<ValType> inputs;`
- `std::optional<ValType> result;` (where empty means `void`)

And check that against the type section. Abort on mismatch.

Reply:
It can be then
```
struct NamedExternalFunction
{
    std::string module;
    std::string name;
    ExternalFunction external_function;
};
```

Reply:
Yeah I guess it could, but that's quite a few different structs to address when declaring these. (From a user perspective.)

Reply:
Alternative is to use a string (similar to what wasm3 and emscripten is doing): parameters followed by colon followed by return types.

e.g. `ii:v` would mean it takes two 32-bit inputs and returns void.

The same in wasm3 is `v(ii)` and `iiv` in emscripten.